tensorflow_estimator/python/estimator/canned/linear.py

```diff
@@ -157,7 +157,8 @@ def _prune_and_unique_sparse_ids(self, id_weight_pair):
           [array_ops.shape(id_tensor.indices)[0]], dtypes.float32)
 
     example_ids = array_ops.reshape(id_tensor.indices[:, 0], [-1])
-    flat_ids = math_ops.to_int64(array_ops.reshape(id_tensor.values, [-1]))
+    flat_ids = math_ops.cast(
+        array_ops.reshape(id_tensor.values, [-1], dtype=dtypes.int64))
```

Comment:
Internal Test failures : `in _prune_and_unique_sparse_ids
    array_ops.reshape(id_tensor.values, [-1], dtype=dtypes.int64))
TypeError: reshape() got an unexpected keyword argument 'dtype'  `

Reply:
Missed this on review, but there should be a `)` immediately after `[-1]` and only one `)` after `int64`.

Thanks for catching it out. @fsx950223 can you do the change please?

Reply:
Fixed